plugins/net_plugin/include/eosio/net_plugin/auto_bp_peering.hpp

```diff
@@ -0,0 +1,226 @@
+   }
+
+   class neighbor_finder_type {
+
+      const config_t&                               config;
+      const std::vector<chain::producer_authority>& schedule;
+      chain::flat_set<std::size_t>                  my_schedule_indices;
+
+    public:
+      neighbor_finder_type(const config_t& config,
+                           const std::vector<chain::producer_authority>& schedule)
+          : config(config), schedule(schedule) {
+         for (auto account : config.my_bp_accounts) {
+            auto itr = std::find_if(schedule.begin(), schedule.end(),
+                                    [account](auto& e) { return e.producer_name == account; });
+            if (itr != schedule.end())
+               my_schedule_indices.insert(itr - schedule.begin());
+         }
+      }
+
+      void add_neighbors_with_distance(chain::flat_set<account_name>& result, int distance) const {
+         for (auto schedule_index : my_schedule_indices) {
+            auto i = (schedule_index + distance) % schedule.size();
+            if (!my_schedule_indices.count(i)) {
+               auto name = schedule[i].producer_name;
+               if (config.bp_peer_addresses.count(name))
+                  result.insert(name);
+            }
+         }
+      }
+
+      flat_set<account_name> downstream_neighbors() const {
+         chain::flat_set<account_name> result;
+         for (int i = 0; i < proximity_count; ++i) { add_neighbors_with_distance(result, i + 1); }
+         return result;
+      }
+
+      void add_upstream_neighbors(chain::flat_set<account_name>& result) const {
+         for (int i = 0; i < proximity_count; ++i) { add_neighbors_with_distance(result, -1 - i); }
+      }
+
+      flat_set<account_name> neighbors() const {
+         flat_set<account_name> result = downstream_neighbors();
+         add_upstream_neighbors(result);
+         return result;
+      }
+   };
+
+ public:
+   const static std::size_t proximity_count = 2;
+
+   bool auto_bp_peering_enabled() const { return config.bp_peer_addresses.size() && config.my_bp_accounts.size(); }
+
+   // Only called at plugin startup
+   void set_producer_accounts(const std::set<account_name>& accounts) {
+      config.my_bp_accounts.insert(accounts.begin(), accounts.end());
+   }
+
+   // Only called at plugin startup
+   void set_bp_peers(const std::vector<std::string>& peers) {
+      try {
+         for (auto& entry : peers) {
+            auto comma_pos = entry.find(',');
+            EOS_ASSERT(comma_pos != std::string::npos, chain::plugin_config_exception,
+                       "auto-bp-peer must consists an account name and server address separated by a comma token");
+            auto         addr = entry.substr(comma_pos + 1);
+            account_name account(entry.substr(0, comma_pos));
+
+            config.bp_peer_addresses[account] = addr;
+            config.bp_peer_accounts[addr]     = account;
+         }
+      } catch (eosio::chain::name_type_exception&) {
```

Comment:
This try-catch could be inside the for loop so it could report which account string(s) are invalid.

Reply:
I believe https://github.com/AntelopeIO/leap/blob/48a4d871db92fa00767490994aab526f4c0db857/libraries/chain/name.cpp#L13 already report the invalid string